authz/grpc_audit_end2end_test.go

```diff
@@ -0,0 +1,304 @@
+package authz
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"io"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/authz/audit"
+	"google.golang.org/grpc/credentials"
+	testgrpc "google.golang.org/grpc/interop/grpc_testing"
+	testpb "google.golang.org/grpc/interop/grpc_testing"
+	"google.golang.org/grpc/testdata"
+
+	_ "google.golang.org/grpc/authz/audit/stdout"
+)
+
+type testServer struct {
+	testgrpc.UnimplementedTestServiceServer
+}
+
+func (s *testServer) UnaryCall(ctx context.Context, req *testpb.SimpleRequest) (*testpb.SimpleResponse, error) {
+	return &testpb.SimpleResponse{}, nil
+}
+
+type statAuditLogger struct {
+	Stat         map[bool]int
```

Comment:
Let's name this something more meaningful - from the impl of `Log` it looks like it is a count of the Authorized vs. Unauthorized calls, mapped to `true` and `false` respectively?

Also, I think the map entries would be more clear if they were just string keys like `allow` and `deny` instead of true and false

Reply:
I renamed it plus added comments but I'd prefer to keep it as map[bool]. It just fits more naturally to Event.Authorized field

Reply:
This is good with me now